fedimint-core/src/db/mod.rs

```diff
@@ -358,7 +467,14 @@ impl<'a> IDatabaseTransaction<'a> for ModuleDatabaseTransaction<'a> {
         Box::pin(stream::iter(stream))
     }
 
-    async fn commit_tx(self: Box<Self>) -> Result<()> {
+    async fn raw_remove_by_prefix(&mut self, key_prefix: &[u8]) -> Result<()> {
+        self.dbtx
+            .with_module_prefix(self.prefix)
+            .raw_remove_by_prefix(key_prefix)
+            .await
+    }
+
+    async fn commit_tx(self) -> Result<()> {
```

Comment:
all internal wrapper would work with and implement `ISingleUseDatabaseTransaction` directly, IMO

Reply:
Yeah, I'm also wondering if this wouldn't be simpler. A bit more redundancy, but one less thing in a common code. Some db trait implementation crates might not even need an `Option`.

Reply:
Changed all wrappers to implement `ISingleUseDatabaseTransaction`